[FSTORE-494] add documentation how to add new filters and/or join new FGs to feature view query (#147)
logicalclocks/logicalclocks.github.io: docs/user_guides/fs/feature_view/query.md

        .join(locationFg.selectAll(), Seq("location_id"), Seq("id"), "left")
        .filter(rainFg.getFeature("location_id").eq(10)))
    ```

#### Joins and/or Filters on feature view query

Exiting feature view query can be extended with new joins and/or new filters. However, this operation
will not update feature view metadata and persist it with new query.

=== "Python"
    ```python
    fs = ...

>     wind_speed_fg = fs.get_feature_group(name=wind_speed, version=1)

Reviewer: ```suggestion
    wind_speed_fg = fs.get_feature_group(name="wind_speed", version=1)
```

>     rain_fg = fs.get_feature_group(name=rain, version=1)

Reviewer: ```suggestion
    rain_fg = fs.get_feature_group(name="rain_fg", version=1)
```

    feature_view = fs.get_feature_view(“rain_dataset”, version=1)
    feature_view.query\
    .join(wind_speed_fg.select_all())\
    .filter((rain_fg.location_id == 54)
    ```

!!! warning
    Evey join/filter operation applied to an existing feature view query instance will update its state and accumulate.
    To successfully apply new join/filter logic it is recommended to refresh query instance by re-fetching it:

=== "Python"
    ```python
    fs = ...

>     wind_speed_fg = fs.get_feature_group(name=wind_speed, version=1)

Reviewer: ```suggestion
    wind_speed_fg = fs.get_feature_group(name="wind_speed", version=1)
```

>     sollar_irradiance_fg = fs.get_feature_group(name=sollar irradiance, version=1)

Reviewer: use `wind_speed_fg` here to be consistent?

>     rain_fg = fs.get_feature_group(name=rain, version=1)

Reviewer: ```suggestion
    rain_fg = fs.get_feature_group(name="rain_fg", version=1)
```

    # fetch new feature view and its query instance
    feature_veiw = fs.get_feature_view(“rain_dataset”, version=1)

    # apply join/filter logic based on location and wind speed

>     query = feature_veiw.query.join(wind_speed_fg.select_all())

Reviewer: ```suggestion
    feature_veiw.query.join(wind_speed_fg.select_all()).filter((rain_fg.location_id == 54)
```
can be simplified in one line.

    query.filter((rain_fg.location_id == 54)

    # to apply new logic indendent of location and wind speed from above 
    # re-fetch new feature view and its query instance
    feature_veiw = fs.get_feature_view(“rain_dataset”, version=1)

    # apply new join/filter logic based on sollar irradiance

>     query = feature_veiw.query.join(sollar_irradiance_fg.select_all())

Reviewer: can be in one line also.

        .join(locationFg.selectAll(), Seq("location_id"), Seq("id"), "left")
        .filter(rainFg.getFeature("location_id").eq(10)))
    ```

#### Joins and/or Filters on feature view query

> Exiting feature view query can be extended with new joins and/or new filters. However, this operation
> will not update feature view metadata and persist it with new query.

Reviewer 2: ```suggestion
The query retrieved from a feature view can be extended with new joins and/or new filters.
However, this operation will not update the metadata and persist the updated query of the feature view itself. This query can then be used to create a new feature view.
```

=== "Python"
    ```python
    fs = ...
    wind_speed_fg = fs.get_feature_group(name="wind_speed_fg", version=1)
    rain_fg = fs.get_feature_group(name="rain_fg", version=1)
    feature_view = fs.get_feature_view(“rain_dataset”, version=1)

>     feature_view.query\
>     .join(wind_speed_fg.select_all())\
>     .filter((rain_fg.location_id == 54)

Reviewer 2: ```suggestion
    feature_view.query \
        .join(wind_speed_fg.select_all()) \
        .filter((rain_fg.location_id == 54)
```

    ```

=== "Scala"
    ```scala
    val fs = ...
    val windSpeedFg = fs.getFeatureGroup("wind_speed_fg", 1)
    val rainFg = fs.getFeatureGroup("rain_fg", 1)
    val featureView = fs.getFeatureView(“rain_dataset”, 1)

>     featureView.getQuery().
>         join(windSpeedFg.selectAll()).
>         filter(rainFg.getFeature("location_id").eq(54))

Reviewer 2: Not sure if this works in jupyter, but it would be much better to read.
```suggestion
    featureView.getQuery()
        .join(windSpeedFg.selectAll())
        .filter(rainFg.getFeature("location_id").eq(54))
```

    ```

!!! warning

>     Evey join/filter operation applied to an existing feature view query instance will update its state and accumulate.
>     To successfully apply new join/filter logic it is recommended to refresh query instance by re-fetching it:

Reviewer 2: ```suggestion
    Every join/filter operation applied to an existing feature view query instance will update its state and accumulate.
    To successfully apply new join/filter logic it is recommended to refresh the query instance by re-fetching the feature view:
```

=== "Python"
    ```python
    fs = ...
    wind_speed_fg = fs.get_feature_group(name="wind_speed_fg", version=1)
    solar_irradiance_fg = fs.get_feature_group(name="solar_irradiance_fg", version=1)
    rain_fg = fs.get_feature_group(name="rain_fg", version=1)

    # fetch new feature view and its query instance

>     feature_veiw = fs.get_feature_view(“rain_dataset”, version=1)

Reviewer 2: ```suggestion
    feature_view = fs.get_feature_view(“rain_dataset”, version=1)
```

    # apply join/filter logic based on location and wind speed

>     feature_veiw.query.join(wind_speed_fg.select_all())\

Reviewer 2: ```suggestion
    feature_view.query.join(wind_speed_fg.select_all()) \
```

        .filter((rain_fg.location_id == 54)

    # to apply new logic independent of location and wind speed from above 
    # re-fetch new feature view and its query instance

>     feature_veiw = fs.get_feature_view(“rain_dataset”, version=1)

Reviewer 2: ```suggestion
    feature_view = fs.get_feature_view(“rain_dataset”, version=1)
```

    # apply new join/filter logic based on solar irradiance

>     feature_veiw.query.join(solar_irradiance_fg.select_all())\

Reviewer 2: ```suggestion
    feature_view.query.join(solar_irradiance_fg.select_all()) \
```

        .filter(solar_irradiance_fg.location_id == 28)
    ```

=== "Scala"
    ```scala
    fs = ...
    windSpeedFg = fs.getFeatureGroup("wind_speed_fg", 1)
    solarIrradianceFg = fs.getFeatureGroup("solar_irradiance_fg", 1)
    rainFg = fs.getFeatureGroup("rain_fg", 1)
    
    // fetch new feature view and its query instance

>     val featureVeiw = fs.getFeatureView(“rain_dataset”, version=1)

Reviewer 2: ```suggestion
    val featureView = fs.getFeatureView(“rain_dataset”, version=1)
```

    
    // apply join/filter logic based on location and wind speed

>     featureVeiw.getQuery.join(windSpeedFg.selectAll()).

Reviewer 2: ```suggestion
    featureView.getQuery.join(windSpeedFg.selectAll()).
```

>     featureView.getQuery.join(windSpeedFg.selectAll()).
>         filter(rainFg.getFeature("location_id").eq(54))

Reviewer 2: ```suggestion
    featureView.getQuery.join(windSpeedFg.selectAll())
        .filter(rainFg.getFeature("location_id").eq(54))
```

    
    // to apply new logic independent of location and wind speed from above 
    // re-fetch new feature view and its query instance

>     val featureVeiw = fs.getFeatureView(“rain_dataset”, 1)

Reviewer 2: ```suggestion
    val featureView = fs.getFeatureView(“rain_dataset”, 1)
```

    
    // apply new join/filter logic based on solar irradiance

>     featureVeiw.getQuery.join(solarIrradianceFg.selectAll()).

Reviewer 2: ```suggestion
    featureView.getQuery.join(solarIrradianceFg.selectAll()).
```

>     featureView.getQuery.join(solarIrradianceFg.selectAll()).
>         filter(solarIrradianceFg.getFeature("location_id").eq(28))

Reviewer 2: ```suggestion
    featureView.getQuery.join(solarIrradianceFg.selectAll())
        .filter(solarIrradianceFg.getFeature("location_id").eq(28))
```
